devito/function.py

```diff
@@ -456,6 +456,56 @@ def _arg_check(self, args, intervals):
     # Pickling support
     _pickle_kwargs = AbstractCachedFunction._pickle_kwargs + ['staggered']
 
+    def __add__(self, other):
```

Comment:
maybe we don't need any of these methods if we make TensorFunction subclass Differentiable? because that's what we probably want right?

Reply:
That's what I want yes, just haven't gotten a way to get the `hasattr/getattr` and commen stuff (`dtype, spcae_order,...)` to work properly yet